backend/src/services/textGenService.ts

```diff
@@ -17,47 +16,27 @@ const sendToProxy = async (json: Prompt): Promise<Gpt3Response> => {
     //Send to proxy if environment is 'openai'
     //Otherwise, expect that we are using the dummy
     //Currently, both expect the service available at port 8080
-    if (process.env.ENVIRONMENT === 'openai') {
-        try {
-            //send to real proxy here, response should be of type Gpt3Response
-            const response = await axios.post<Gpt3Response>(
-                'http://localhost:8080',
-                json
-            );
+    try {
+        if (process.env.ENVIRONMENT === 'openai') {
+            //send to the real proxy here
+            const proxyURL = process.env.PROXYURL
+                ? process.env.PROXYURL
+                : 'not_found';
+            const response = await axios.post<Gpt3Response>(proxyURL, json);
+            return response.data;
+        } else {
+            //send to the dummy here
+            const dummyURL = process.env.DUMMYURL
+                ? process.env.DUMMYURL
+                : 'not_found';
+            const response = await axios.post<Gpt3Response>(dummyURL, json);
             return response.data;
-        } catch (e) {
-            throw new ProxyError(
-                'Error when processing response from proxy\n' +
-                    (e as SyntaxError).message
-            );
-        }
-    } else {
-        try {
-            //dummy
-            const response = await axios.post<DummyResponse>(
-                'http://localhost:8080',
-                json
-            );
-
-            const { gpt, debug } = response.data;
-            //Do some logging, then return the actual gpt3 response with stuff appended
-            console.log(
-                `Replied to prompt:\n"${debug.prompt.prompt}"\nDate: ${debug.date}`
-            );
-
-            const debugText = `This is a demo prompt generated using a dummy backend
-Date: ${debug.date}
-Prompt: ${JSON.stringify(debug.prompt)}
-id: ${getRndString(32)}`;
-
-            gpt.choices[0].text = debugText + gpt.choices[0].text;
-            return gpt;
-        } catch (e) {
-            throw new ProxyError(
-                'Error when processing response from dummy\n' +
-                    (e as SyntaxError).message
-            );
         }
```

Comment:
The if else statement from line 20 to line 34 has unnecessary repetition.
This could be replaced with following code:

const proxyURL = process.env.ENVIRONMENT === 'openai'
    ? process.env.PROXYURL
    : process.env.DUMMYURL;
const response = await axios.post<Gpt3Response>(proxyURL, json);
return response.data;

If typescript complains that proxyURL is not a string I would try

const proxyURL = process.env.ENVIRONMENT === 'openai'
    ? process.env.PROXYURL as string
    : process.env.DUMMYURL as string;

or

const proxyURL = (process.env.ENVIRONMENT === 'openai'
    ? process.env.PROXYURL
    : process.env.DUMMYURL) || 'not_found';

For future reference, the following code:

const proxyURL = process.env.PROXYURL
    ? process.env.PROXYURL
    : 'not_found';

could be simply written as:

const proxyURL = process.env.PROXYURL || 'not_found';

but that is not required here since it makes no difference in our case wethter the prodyURL is undefined or 'not_found'. In both cases we will be getting an error.